File: app/src/main/java/com/gfq/dialog/expand/ChooseSalaryRangeDialog.java
package com.gfq.dialog.expand;

import android.content.Context;
import android.graphics.Color;

import com.contrarywind.adapter.WheelAdapter;
import com.contrarywind.listener.OnItemSelectedListener;
import com.gfq.dialog.R;
import com.gfq.dialog.base.BaseBottomRoundDialog;
import com.gfq.dialog.base.BaseRoundDialog;
import com.gfq.dialog.base.DialogType;
import com.gfq.dialog.base.GDialog;
import com.gfq.dialog.databinding.DialogChooseSalaryRangeBinding;

import java.util.ArrayList;
import java.util.List;

/**
 * @created GaoFuq
 * @Date 2020/7/22 16:32
 * @Descaption
 */
public class ChooseSalaryRangeDialog {

    private Context context;
    private DialogType dialogType;
    private int minSalary = 1;
    private int maxSalary = 50;
    private String label = "k";//单位
    private int wvTextColor = Color.parseColor("#999999");
    private int wvTextColorCenter = Color.parseColor("#333333");
    private int wvTextSize = 16;
    private GDialog dialog;
    private DialogChooseSalaryRangeBinding binding;
    private WheelAdapter<String> wvMinAdapter;
    private WheelAdapter<String> wvMaxAdapter;
    private ArrayList<Integer> minSalaryList;
    private ArrayList<Integer> maxSalaryList;
    private int selectedMinSalary;
    private int selectedMaxSalary;


    public ChooseSalaryRangeDialog(Context context, DialogType dialogType) {
        this.context = context;
        this.dialogType = dialogType;
        initData();
        initDialog();
    }

    public ChooseSalaryRangeDialog(Context context, DialogType dialogType, int minSalary, int maxSalary, String label) {
        this.context = context;
        this.dialogType = dialogType;
        this.minSalary = minSalary;
        this.maxSalary = maxSalary;
        this.label = label;
        initData();
        initDialog();
    }

    private void initData() {
        if (minSalaryList == null) {
            minSalaryList = new ArrayList<>();
        }
        if (maxSalaryList == null) {
            maxSalaryList = new ArrayList<>();
        }

        for (int i = minSalary; i < maxSalary + 1; i++) {
            minSalaryList.add(i);
        }

        for (int i = minSalary + 1; i < maxSalary + 1; i++) {
            maxSalaryList.add(i);
        }

        wvMinAdapter = new SalaryAdapter(minSalaryList);
        wvMaxAdapter = new SalaryAdapter(maxSalaryList);


    }


    private void initDialog() {
        if (context != null) {
            if (dialog == null) {
                if (dialogType == DialogType.normal) {
                    dialog = new BaseRoundDialog<DialogChooseSalaryRangeBinding>(context) {
                        @Override
                        protected int layout() {
                            return R.layout.dialog_choose_salary_range;
                        }

                        @Override
                        protected void bindView(DialogChooseSalaryRangeBinding dialogBinding) {
                            bindDialogView(dialogBinding);
                        }
                    };
                } else if (dialogType == DialogType.bottom) {
                    dialog = new BaseBottomRoundDialog<DialogChooseSalaryRangeBinding>(context) {
                        @Override
                        protected int layout() {
                            return R.layout.dialog_choose_salary_range;
                        }

                        @Override
                        protected void bindView(DialogChooseSalaryRangeBinding dialogBinding) {
                                bindDialogView(dialogBinding);
                        }
                    };
                }


            }
        }
    }

    private void bindDialogView(DialogChooseSalaryRangeBinding dialogBinding) {

        binding.wvMin.setAdapter(wvMinAdapter);
        binding.wvMax.setAdapter(wvMaxAdapter);

        binding.wvMin.setCyclic(false);
        binding.wvMax.setCyclic(false);
        binding.wvMin.setCurrentItem(0);
        binding.wvMax.setCurrentItem(0);

        setWheelViewDefStyle(wvTextColor, wvTextColorCenter, wvTextSize);
        setLineHeight(3);
        isCenterLabel(false);

        binding.wvMin.setOnItemSelectedListener(new OnItemSelectedListener() {
            @Override
            public void onItemSelected(int index) {
                selectedMinSalary = minSalaryList.get(index);
                maxSalaryList.clear();
                if (selectedMinSalary == maxSalary) {
                    maxSalaryList.add(selectedMinSalary);
                    binding.wvMax.setAdapter(new SalaryAdapter(maxSalaryList));
                    return;
                }
                if(selectedMinSalary>=selectedMaxSalary)
                    for (int i = selectedMinSalary + 1; i < maxSalary + 1; i++) {
                        maxSalaryList.add(i);
                    }
                binding.wvMax.setAdapter(new SalaryAdapter(maxSalaryList));
                binding.wvMax.setCurrentItem(0);

            }
        });
        binding.wvMax.setOnItemSelectedListener(new OnItemSelectedListener() {
            @Override
            public void onItemSelected(int index) {
                selectedMaxSalary = maxSalaryList.get(index);
            }
        });

        binding.tvConfirm.setOnClickListener(v -> {
            if (onConfirmListener != null) {
                onConfirmListener.onConfirm(selectedMinSalary + label, selectedMaxSalary + label);
            }
            dialog.dismiss();
        });

        binding.tvCancel.setOnClickListener(v -> dialog.dismiss());
    }


    private void setWheelViewDefStyle(int textColor, int textColorCenter, int textSize) {
        binding.wvMin.setTextColorOut(textColor);
        binding.wvMin.setTextColorCenter(textColorCenter);
        binding.wvMin.setTextSize(textSize);
        binding.wvMin.setLabel(label);

        binding.wvMax.setTextColorOut(textColor);
        binding.wvMax.setTextColorCenter(textColorCenter);
        binding.wvMax.setTextSize(textSize);
        binding.wvMax.setLabel(label);
    }

    //是否只显示中间的Label  默认为true
    public void isCenterLabel(boolean b) {
        binding.wvMin.isCenterLabel(b);
        binding.wvMax.isCenterLabel(b);
    }

    public void setWheelViewStyle(int textColor, int textColorCenter, int textSize) {
        setWheelViewDefStyle(textColor, textColorCenter, textSize);
    }

    public void setTitleStyle(String text, int textColor, int textSize) {
        binding.tvTitle.setText(text);
        binding.tvTitle.setTextColor(textColor);
        binding.tvTitle.setTextSize(textSize);
    }

    public void setConfirmStyle(String text, int textColor, int textSize) {
        binding.tvConfirm.setText(text);
        binding.tvConfirm.setTextColor(textColor);
        binding.tvConfirm.setTextSize(textSize);
    }

    public void setCancelStyle(String text, int textColor, int textSize) {
        binding.tvCancel.setText(text);
        binding.tvCancel.setTextColor(textColor);
        binding.tvCancel.setTextSize(textSize);
    }

    public void setLineHeight(float lineHeight) {
        binding.wvMin.setLineSpacingMultiplier(lineHeight);
        binding.wvMax.setLineSpacingMultiplier(lineHeight);
    }

    public void show() {
        if (dialog != null) {
            dialog.show();
        }
    }


    public void dismiss(){
        dialog.dismiss();
    }

    public interface OnConfirmListener {
        void onConfirm(String minSalary, String maxSalary);
    }

    private OnConfirmListener onConfirmListener;

    public void setOnConfirmListener(OnConfirmListener onConfirmListener) {
        this.onConfirmListener = onConfirmListener;
    }

    static class SalaryAdapter implements WheelAdapter<String> {

        private List<Integer> list;

        public SalaryAdapter(List<Integer> list) {
            this.list = list;
        }

        @Override
        public int getItemsCount() {
            return list.size();
        }

        @Override
        public String getItem(int index) {
            return list.get(index) + "";
        }

        @Override
        public int indexOf(String o) {
            return 0;
        }
    }
}

File: app/src/main/java/com/gfq/dialog/expand/ChooseSalaryRangeDialog.kt
package com.gfq.dialog.expand

import android.content.Context
import android.graphics.Color
import android.view.View
import com.contrarywind.adapter.WheelAdapter
import com.contrarywind.listener.OnItemSelectedListener
import com.gfq.dialog.R
import com.gfq.dialog.base.BaseDialog
import com.gfq.dialog.databinding.DialogChooseSalaryRangeBinding
import java.util.*

/**
 * @created GaoFuq
 * @Date 2020/7/22 16:32
 * @Descaption
 */
class ChooseSalaryRangeDialog(minSalary: Int, maxSalary: Int, label: String) : BaseDialog<DialogChooseSalaryRangeBinding>() {
    var minSalary = 1
    var maxSalary = 50
    var label = "k" //单位
    var wvTextColor = Color.parseColor("#999999")
    var wvTextColorCenter = Color.parseColor("#333333")
    var wvTextSize = 16
    private var wvMinAdapter: WheelAdapter<String>? = null
    private var wvMaxAdapter: WheelAdapter<String>? = null
    private var minSalaryList = mutableListOf<Int>()
    private var maxSalaryList =mutableListOf<Int>()
    private var selectedMinSalary = 0
    private var selectedMaxSalary = 0
    override fun layout(): Int {
        return R.layout.dialog_choose_salary_range
    }

    override fun bindView() {
        initData()
        bindDialogView()
    }

    private fun initData() {
        for (i in minSalary until maxSalary + 1) {
            minSalaryList.add(i)
        }
        for (i in minSalary + 1 until maxSalary + 1) {
            maxSalaryList.add(i)
        }
        wvMinAdapter = SalaryAdapter(minSalaryList)
        wvMaxAdapter = SalaryAdapter(maxSalaryList)
    }

    private fun bindDialogView() {
        dgBinding!!.wvMin.adapter = wvMinAdapter
        dgBinding!!.wvMax.adapter = wvMaxAdapter
        dgBinding!!.wvMin.setCyclic(false)
        dgBinding!!.wvMax.setCyclic(false)
        dgBinding!!.wvMin.currentItem = 0
        dgBinding!!.wvMax.currentItem = 0
        setWheelViewDefStyle(wvTextColor, wvTextColorCenter, wvTextSize)
        setLineHeight(3f)
        isCenterLabel(false)
        dgBinding!!.wvMin.setOnItemSelectedListener(OnItemSelectedListener { index ->
            selectedMinSalary = minSalaryList!![index]
            maxSalaryList!!.clear()
            if (selectedMinSalary == maxSalary) {
                maxSalaryList!!.add(selectedMinSalary)
                dgBinding!!.wvMax.adapter = SalaryAdapter(maxSalaryList)
                return@OnItemSelectedListener
            }
            if (selectedMinSalary >= selectedMaxSalary) {
                for (i in selectedMinSalary + 1 until maxSalary + 1) {
                    maxSalaryList!!.add(i)
                }
            }
            dgBinding!!.wvMax.adapter = SalaryAdapter(maxSalaryList)
            dgBinding!!.wvMax.currentItem = 0
        })
        dgBinding!!.wvMax.setOnItemSelectedListener { index -> selectedMaxSalary = maxSalaryList!![index] }
        dgBinding!!.tvConfirm.setOnClickListener { v: View? ->
            if (onConfirmListener != null) {
                onConfirmListener!!.onConfirm(selectedMinSalary.toString() + label, selectedMaxSalary.toString() + label)
            }
            dismiss()
        }
        dgBinding!!.tvCancel.setOnClickListener { v: View? -> dismiss() }
    }

    private fun setWheelViewDefStyle(textColor: Int, textColorCenter: Int, textSize: Int) {
        dgBinding!!.wvMin.setTextColorOut(textColor)
        dgBinding!!.wvMin.setTextColorCenter(textColorCenter)
        dgBinding!!.wvMin.setTextSize(textSize.toFloat())
        dgBinding!!.wvMin.setLabel(label)
        dgBinding!!.wvMax.setTextColorOut(textColor)
        dgBinding!!.wvMax.setTextColorCenter(textColorCenter)
        dgBinding!!.wvMax.setTextSize(textSize.toFloat())
        dgBinding!!.wvMax.setLabel(label)
    }

    //是否只显示中间的Label  默认为true
    fun isCenterLabel(b: Boolean) {
        dgBinding!!.wvMin.isCenterLabel(b)
        dgBinding!!.wvMax.isCenterLabel(b)
    }

    fun setWheelViewStyle(textColor: Int, textColorCenter: Int, textSize: Int) {
        setWheelViewDefStyle(textColor, textColorCenter, textSize)
    }

    fun setTitleStyle(text: String?, textColor: Int, textSize: Int) {
        dgBinding!!.tvTitle.text = text
        dgBinding!!.tvTitle.setTextColor(textColor)
        dgBinding!!.tvTitle.textSize = textSize.toFloat()
    }

    fun setConfirmStyle(text: String?, textColor: Int, textSize: Int) {
        dgBinding!!.tvConfirm.text = text
        dgBinding!!.tvConfirm.setTextColor(textColor)
        dgBinding!!.tvConfirm.textSize = textSize.toFloat()
    }

    fun setCancelStyle(text: String?, textColor: Int, textSize: Int) {
        dgBinding!!.tvCancel.text = text
        dgBinding!!.tvCancel.setTextColor(textColor)
        dgBinding!!.tvCancel.textSize = textSize.toFloat()
    }

    fun setLineHeight(lineHeight: Float) {
        dgBinding!!.wvMin.setLineSpacingMultiplier(lineHeight)
        dgBinding!!.wvMax.setLineSpacingMultiplier(lineHeight)
    }

    interface OnConfirmListener {
        fun onConfirm(minSalary: String?, maxSalary: String?)
    }

    private var onConfirmListener: OnConfirmListener? = null
    fun setOnConfirmListener(onConfirmListener: OnConfirmListener?) {
        this.onConfirmListener = onConfirmListener
    }

    internal class SalaryAdapter(private val list: List<Int>) : WheelAdapter<String> {
        override fun getItemsCount(): Int {
            return list.size
        }

        override fun getItem(index: Int): String {
            return list[index].toString() + ""
        }

        override fun indexOf(o: String): Int {
            return 0
        }
    }

    init {
        this.minSalary = minSalary
        this.maxSalary = maxSalary
        this.label = label
    }
}